### DISCUSSION TITLE
Larastan types in PHPStorm

### DISCUSSION THREAD

**Discussion by jivanf** [larastan/larastan#1661] [Q&A]
When using Larastan types in PHPStorm, PHPStorm is unable to recognize them and shows 'undefined class' errors.

![image](https://github.com/nunomaduro/larastan/assets/25129083/11e55d68-ee53-47de-ab92-aadf220447ad)

Fortunately, this isn't a big issue because it's only on the IDE side (nothing else is affected) and the error goes away when adding `@noinspection PhpUndefinedClassInspection`. However, is it possible to fix this error instead of suppressing it with this tag? It'd be nice to not have to do this whenever I need to use a Larastan type.

**Comment by szepeviktor** [comment]
Please see https://youtrack.jetbrains.com/issue/WI-55886

You could go like
```php
/**
 * @phpstan-var view-string
 * @var string
 */
```

**Reply by jivanf** [reply]
Sadly that doesn't seem to do the trick, `view-string` is still not recognized even when it's used with `@phpstan-var`. I looked at the issue and they fixed it by installing the Psalm plugin. In my case, I already have the PHPStan plugin installed and configured, so that's not the problem.
![image](https://github.com/nunomaduro/larastan/assets/25129083/a52ad8ce-2caf-4dee-acaa-40bc245e38e7)

**Reply by szepeviktor** [reply]
What does you IDE say to `@phpstan-var`?

**Reply by szepeviktor** [reply]
I see! The PHPStan plugin does not understand `view-string`.
